nomad/job_endpoint_hook_connect.go

```diff
@@ -434,7 +447,14 @@ func gatewayBindAddressesIngressForBridge(ingress *structs.ConsulIngressConfigEn
 	return addresses
 }
 
-func newConnectGatewayTask(prefix, service string, netHost bool) *structs.Task {
+func newConnectGatewayTask(prefix, service string, netHost, customizedTls bool) *structs.Task {
+	constraints := structs.Constraints{
+		connectGatewayVersionConstraint(),
+		connectListenerConstraint(),
+	}
+	if customizedTls {
+		constraints = append(constraints, connectGatewayTLSVersionConstraint())
+	}
```

Comment:
It would be good to have a new test to validate the `groupConnectHook` logic around this conditionally added constraint.